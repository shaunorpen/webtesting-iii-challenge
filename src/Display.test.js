import React from 'react';
import * as rtl from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Display from './Display';

afterEach(rtl.cleanup);

let wrapper;

beforeEach(() => {
  wrapper = rtl.render(<Display />);
})

describe('Display component:', () => {
    it('renders correctly', () => {
        wrapper.debug();
        expect(wrapper.container).toMatchSnapshot();
    });

    it('displays if gate is open/closed and if it is locked/unlocked', () => {

    });

    it('displays Closed if the closed prop is true and Open if otherwise', () => {

    });

    it('displays Locked if the locked prop is true and Unlocked if otherwise', () => {

    });

    it('when locked or closed use the red-led class', () => {

    });

    it('when unlocked or open use the green-led class', () => {

    });
});