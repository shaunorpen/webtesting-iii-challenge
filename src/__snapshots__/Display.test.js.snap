// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component: renders correctly 1`] = `
<div>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
    >
      Open
    </div>
  </div>
</div>
`;

exports[`Display component: renders correctly 2`] = `
<div>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
    >
      Unlocked
    </div>
    <div
      class="led red-led"
    >
      Closed
    </div>
  </div>
</div>
`;

exports[`Display component: renders correctly 3`] = `
<div>
  <div
    class="display panel"
  >
    <div
      class="led red-led"
    >
      Locked
    </div>
    <div
      class="led green-led"
    >
      Open
    </div>
  </div>
</div>
`;

exports[`Display component: renders correctly 4`] = `
<div>
  <div
    class="display panel"
  >
    <div
      class="led red-led"
    >
      Locked
    </div>
    <div
      class="led red-led"
    >
      Closed
    </div>
  </div>
</div>
`;
